framework/src/postprocessors/MemoryUsage.C

```diff
@@ -0,0 +1,145 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "MemoryUsage.h"
+
+#include <array>
+#include <unistd.h>
+
+#ifdef __APPLE__
+#include <mach/task.h>
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
+
+template<>
+InputParameters validParams<MemoryUsage>()
+{
+  InputParameters params = validParams<GeneralPostprocessor>();
+  params.addClassDescription("Memory usage statistics for the running simulation.");
+  MooseEnum mem_type("virtual_memory physical_memory page_faults", "virtual_memory");
+  params.addParam<MooseEnum>("mem_type", mem_type, "Memory metric to report.");
+  MooseEnum value_type("total average max_process min_processs", "total");
+  params.addParam<MooseEnum>("value_type", value_type, "Aggregation method to apply to the requested memory metric.");
+  params.addParam<bool>("report_peak_value", true, "If the postprocessor is executed more than one during a time step, report the aggregated peak value.");
+  return params;
+}
+
+MemoryUsage::MemoryUsage(const InputParameters & parameters) :
+    GeneralPostprocessor(parameters),
+    _mem_type(getParam<MooseEnum>("mem_type").getEnum<MemType>()),
+    _value_type(getParam<MooseEnum>("value_type").getEnum<ValueType>()),
+    _report_peak_value(getParam<bool>("report_peak_value"))
+{
+}
+
+void
+MemoryUsage::timestepSetup()
+{
+  _peak_value = 0.0;
+}
+
+void
+MemoryUsage::initialize()
+{
+  _value = 0.0;
+}
+
+void
+MemoryUsage::execute()
+{
+  // inspect /proc
+  std::ifstream stat_stream("/proc/self/stat", std::ios_base::in);
+  std::array<unsigned long, 21> val;
```

Comment:
What's the significance of this integer? Can we use a constant expression instead? Possibly below as well where you access the 20th item.